api/src/models/env.rs

```diff
@@ -1,14 +1,21 @@
-use crate::lang::LangValue;
+use crate::{lang::LangValue, schema::environments};
+use diesel::{Identifiable, Queryable};
 use serde::{Deserialize, Serialize};
 
 /// The environment surrounding a program. This is needed both at compile time
 /// and runtime. In the context of the game, this represents one puzzle.
-#[derive(Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, PartialEq, Serialize, Deserialize, Identifiable, Queryable)]
 pub struct Environment {
+    /// Unique ID to refer to this environment
+    pub id: i32,
+
+    // These two need to be i32s because postgres has no unsigned type.
+    // The insertion code and DB should both enforce that they are >= 0.
     /// Maximum number of stacks permitted
-    pub num_stacks: usize,
+    pub num_stacks: i32,
     /// Maximum size of each stack. If None, the capacity is unlimited.
-    pub max_stack_size: Option<usize>,
+    pub max_stack_size: Option<i32>,
```

Comment:
Had to change these to `i32`s, because postgres has no unsigned type so `usize` doesn't can't map to anything in postgres.